src/thor/triplegbuilder.cc

```diff
@@ -481,6 +481,43 @@ void AddTransitNodes(TripLeg_Node* trip_node,
   }
 }
 
+void SetTripEdgeRoadClass(TripLeg_Edge* trip_edge,
+                          const DirectedEdge* directededge,
+                          const GraphTile* graphtile,
+                          GraphReader& graphreader) {
+  trip_edge->set_road_class(GetTripLegRoadClass(directededge->classification()));
+  // If this is a ramp it may have been reclassified in graph enhancer.
+  // To restore the original road class for motorway_links, we check if any of the adjacent edges is
+  // a motorway.
+  if (directededge->use() == Use::kRamp) {
+    const DirectedEdge* opposing_edge = graphreader.GetOpposingEdge(directededge, graphtile);
+    for (const auto& edge : {directededge, opposing_edge}) {
```

Comment:
should probably be `const auto* edge`. type deduction fixes it automatically but to the reader its more illustrative if you mark it with the asterisk

Reply:
Thanks, fixed.